test/conformance/header_test.go

```diff
@@ -83,22 +84,25 @@ func TestShouldHaveHeadersSet(t *testing.T) {
 	headers := ri.Request.Headers
 
 	for header, regex := range expectedHeaders {
-		hv, ok := headers[header]
-		if !ok {
-			t.Errorf("Header %s was not present on request", header)
-			continue
-		}
+		hvl, ok := headers[http.CanonicalHeaderKey(header)]
+		// Check against each value for the header key
+		for _, hv := range hvl {
+			if !ok {
+				t.Errorf("Header %s was not present on request", header)
+				continue
+			}
 
-		switch {
-		case strings.EqualFold(header, "x-forwarded-for"):
-			for _, ip := range strings.Split(hv, ",") {
-				if net.ParseIP(strings.TrimSpace(ip)) == nil {
-					t.Errorf("Header %s has invalid IP: %s", header, ip)
+			switch {
+			case strings.EqualFold(header, "x-forwarded-for"):
+				for _, ip := range strings.Split(hv, ",") {
```

Comment:
So the IPs here won't be as a list?

Reply:
Nope.

I can't link the line, but see line 138 of https://golang.org/src/net/http/request.go

The list is used if the header is set twice.